### PR TITLE
Issue/sling 9881

### PR DESCRIPTION
**Pull request by DominikSuess**

### CODE REVIEW

File: src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java [hunk 1]
```diff
@@ -65,13 +74,77 @@ protected void onJcrRootElement(String uri, String localName, String qName, Attr
...
+                                break;
+                            case PropertyType.DOUBLE:
+                                attributeQName += ":Double";
+                                value = Lists.transform(strValues, new Function<String, Double>() {
+                                   public Double apply(String s) {
+                                      Double res = null;
+                                      try {
+                                          res = Double.parseDouble(s);
+                                      } catch ( NumberFormatException ex ){
+                                          // ignore as nonexisting
```
Comment (rombert) [inline src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java:105]: Are you sure that we should just skip invalid values for doubles and longs and not fail the conversion process?

File: src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java [hunk 2]
```diff
@@ -65,13 +74,77 @@ protected void onJcrRootElement(String uri, String localName, String qName, Attr
 
                     if (attributeValue != null && !attributeValue.isEmpty()) {
                         DocViewProperty property = DocViewProperty.parse(attributeQName, attributeValue);
-
+                        Object value = property.values;
+                        List<String> strValues = Arrays.asList(property.values);
+                        switch (property.type) {
+                            case PropertyType.DATE:
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java:80]: The JCR https://docs.adobe.com/docs/en/spec/javax.jcr/javadocs/jcr-2.0/javax/jcr/PropertyType.html contains more types than explicitly handled here. Are we sure that we cover all the ones we need?

File: src/main/java/org/apache/sling/feature/cpconverter/vltpkg/VaultPackageAssembler.java [hunk 3]
```diff
@@ -105,7 +105,7 @@ private static VaultPackageAssembler create(VaultPackage vaultPackage, Workspace
         Properties properties = new Properties();
         properties.setProperty(PackageProperties.NAME_VERSION,
                                packageProperties.getProperty(PackageProperties.NAME_VERSION)
-                                                             + '-'
+                                                             + '.'
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/cpconverter/vltpkg/VaultPackageAssembler.java:108]: Is this change related?

**Comment by DominikSuess** [comment]
Reverted the switch of separator that was accidentially  commited along but was found to no longer be a concrete issue (fix wouldn't change failing conditions)

File: src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java (hunk 2 above)
Comment (DominikSuess) [reply src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java:80]: @bosschaert I only handle what the sling osgi installer supports and handles - other types wouldn't work in sling:OsgiConfig nodes anyhow. The cnd doesn't constrain it but the other properties woud be handled as String representations in the osgi installer as well.

File: src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java (hunk 1 above)
Comment (DominikSuess) [reply src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java:105]: @rombert  - as those are typed representations (it is a string representation of a  typed property in the end) the only "valid" case of not being a number is an unset value. Therefore I saw no scenario where an explicit handling would do anything here.

Comment (rombert) [reply src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java:105]: Wouldn't the below OSGi config snippet be silently ignored with this change?

```xml
  alertThreshold="{Double}high"
```

?

I think such invalid inputs should halt the conversion process.

Comment (DominikSuess) [reply src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java:105]: @rombert ok - in theory they shouldn't exist but yes you are right if somebody crafted  a broken vault package that may happen - I can make this throw an exception.

Comment (rombert) [reply src/main/java/org/apache/sling/feature/cpconverter/handlers/XmlConfigurationEntryHandler.java]: Great, thanks!

**Comment by rombert** [comment]
@DominikSuess  - can you please force-push a single commit with all the changes? I will then merge

**Comment by kwin** [comment]
@rombert Why not using https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits?

**Comment by rombert** [comment]
@kwin - we settled on a 'squash, rebase and merge' approach which is not possible with the GitHub UI, see the "Merging pull requests from contributors" section in https://cwiki.apache.org/confluence/display/SLING/Using+Git+with+Sling .
